### ISSUE TITLE
aws-lambda: default values discrepancies between TypeScript and Web documentation

### ISSUE THREAD

**Issue by Serede** [aws/aws-cdk#13908]
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.EventSourceMappingOptions.html

There seems to be some inconsistencies in the default values specified for the `EventSourceMappingOptions` interface between [TypeScript source](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-event-sources/lib/stream.ts) and [Web](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.EventSourceMappingOptions.html) versions of the CDK package documentation.

At least, these are the ones that I found:
- `maxRecordAge`
  * Source says [`@default Duration.days(7)`](https://github.com/aws/aws-cdk/blob/f9b67505f3f5bf3f42be028f9d370c4096a9d4dc/packages/%40aws-cdk/aws-lambda-event-sources/lib/stream.ts#L44).
  * Web says *default: infinite or until the record expires*.
  * Empirical default value: `-1` (infinite)
- `retryAttempts`
  * Source says [`@default 10000`](https://github.com/aws/aws-cdk/blob/f9b67505f3f5bf3f42be028f9d370c4096a9d4dc/packages/%40aws-cdk/aws-lambda-event-sources/lib/stream.ts#L54).
  * Web says *default: infinite or until the record expires*.
  * Empirical default value: `-1` (infinite)

Please make default values consistent between TypeScript and Web versions of the documentation as many of us rely on TypeScript LSP for documentation.

---

This is a 📕 documentation issue

**Comment by nija-at** [comment]
Hi @Serede -

You are comparing the wrong artifacts. The typescript source you linked is of `StreamEventSourceProps` and here is the documentation for it - https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda-event-sources.StreamEventSourceProps.html.

**Comment by Serede** [comment]
@nija-at you are right, thank you for clarifying. I still believe there is an issue with the default values, though. Let me explain better.

I created a `DynamoEventSource` like this:
```typescript
    const testrunsEventSource = new DynamoEventSource(testrunsTable, {
      batchSize: 100,
      onFailure: new SqsDlq(deadLetterQueue),
      startingPosition: StartingPosition.TRIM_HORIZON,
    });
```

I see the constructor for `DynamoEventSource` takes `DynamoEventSourceProps` as second argument, which is itself an extension of the `StreamEventSourceProps` you just linked. There, the default values for `StreamEventSourceProps` are:
- `maxRecordAge`. Type: `Duration` (optional, default: `Duration.days(7)`)
- `retryAttempts`. Type: `number` (optional, default: `10000`)

So far so good. However, when I deployed the stack to CloudFormation, **both properties took value `-1` (infinite)**.

I had to change the code like this to make them finite:
```typescript
    const testrunsEventSource = new DynamoEventSource(testrunsTable, {
      batchSize: 100,
      onFailure: new SqsDlq(deadLetterQueue),
      startingPosition: StartingPosition.TRIM_HORIZON,
      maxRecordAge: Duration.minutes(10),
      retryAttempts: 10,
    });
```

I reported the issue because I believe this can cause unintended behavior to other users as it did for me.
